modules/placeholder.js

```diff
@@ -0,0 +1,67 @@
+import Emitter from '../core/emitter';
+import Module from '../core/module';
+
+const className = 'ql-placeholder';
+
+class PlaceHolder extends Module {
+  constructor(quill, options = { content : '', className: className }) {
+    super(quill, options);
+    this.className = typeof options === 'object' ? options.className : className;
+    this.placeHolderText = typeof options === 'object' ? options.content : options;
+    this.bind();
+  }
+
+  bind() {
+    let root = this.quill.root;
+
+    if (!this.hasElements) this.show();
+
+    root.addEventListener('blur', _ => {
+      if (!this.hasElements) this.show();
+    });
+
+    root.addEventListener('focus', _ => {
+      if (!this.hasElements) this.hide();
+    });
+
+    this.quill.on(Emitter.events.TEXT_CHANGE, _ => {
+      if (this.quill.getText().length === 1) {
```

Comment:
`this.quill.getLength()` might be more clear. Also might as well have it be `<= 1` instead of `=== 1`